src/main/java/com/enonic/lib/guillotine/macro/HtmlMacroResult.java

```diff
@@ -0,0 +1,26 @@
+package com.enonic.lib.guillotine.macro;
+
+import java.util.Map;
+
+public class HtmlMacroResult
+{
+    private final String processedAsHtml;
+
+    private final Map<String, Object> processedAsJson;
+
+    HtmlMacroResult( final MacroDecorator macro )
+    {
+        this.processedAsHtml = new MacroEditorSerializer( macro ).serialize();
+        this.processedAsJson = new MacroEditorJsonSerializer( macro ).serialize();
+    }
+
+    public String getProcessedAsHtml()
+    {
+        return processedAsHtml;
```

Comment:
You may save some CPU cycles if make serialisation lazy, because very few need both html and JSON version together

Reply:
I will not do it, because we should have a source before we can read value for specific field.